linkerd/app/src/env.rs

```diff
@@ -125,6 +128,7 @@ pub const ENV_IDENTITY_SVC_BASE: &str = "LINKERD2_PROXY_IDENTITY_SVC";
 pub const ENV_DESTINATION_SVC_BASE: &str = "LINKERD2_PROXY_DESTINATION_SVC";
 
 pub const ENV_HOSTNAME: &str = "HOSTNAME";
+pub const ENV_NAMESPACE: &str = "_pod_ns";
```

Comment:
Strong -1 on this. This couples the proxy to a temp variable used in the injector. The proxy shouldn't know anything about this, or namespaces for that matter. I'd prefer to support an additional env variable that can extend the set of tracing labels.

Reply:
Hmm, Interesting. How does this coupling differ from the ENV coupling we did for the `metadata.labels` file injection, They both use env variables added by the injector, right? why is it temporary?

This could be solved easily by making the proxy injector also, add the `namespace` as a label to the pod, Just want to understand why this is a strong negative :)

Reply:
I'd rather that the proxy not know anything about namespaces at all. The proxy doesn't know about Kubernetes, and namespaces are a kubernetes concept. The only coupling the proxy has with Kubernetes are the default values for the resolution suffix and, now, the path to the metadata labels. I'd prefer that we not default any of these values, to be hoenst.

The proxy exposes a set of `LINKERD2_PROXY_*` environment variables that configure the proxy. The proxy injector sets those, but other things may set them as well, like a test harness, for instance.  The proxy injector uses this environment as an API, basically. The proxy shouldn't have any awareness of how the proxy injector goes about building those values. In this case `_pod_ns` is just part of that implementation detail of how the injector builds identity strings.

I'd prefer that we expose a tracing-specific environment variable that can be used to augment the attributes, rather than specifically exposing a _namespace_ that then goes and does that. The proxy doesn't care about namespaces, it cares about tracing attributes.

Reply:
> In this case _pod_ns is just part of that implementation detail of how the injector builds identity strings.
Yep, That makes sense. `namespace` is not an important detail for the proxy and should not be a dependency. I will update the control-plane PR to also include `namespace` as a label, thus making the tracing attributes path, the only way of adding attributes to traces!